client/app/components/queries/ScheduleDialog.jsx

```diff
@@ -158,104 +127,96 @@ class ScheduleDialog extends React.Component {
-      minute: timeParts ? timeParts[1] : null,
-      dayOfWeek: newSchedule.day_of_week ? WEEKDAYS_SHORT[WEEKDAYS_FULL.indexOf(newSchedule.day_of_week)] : null,
+      hour,
+      minute,
+      dayOfWeek: newSchedule.day_of_week
+        ? WEEKDAYS_SHORT[WEEKDAYS_FULL.indexOf(newSchedule.day_of_week)]
+        : null,
     });
 
-    this.props.updateQuery({
-      schedule: Object.assign(newSchedule, { interval: totalSeconds }),
-    });
+    this.newSchedule = newSchedule;
   };
+
   setCount = (e) => {
     const newCount = e.target.value;
     const totalSeconds = intervalToSeconds(parseInt(newCount, 10), this.state.interval);
     this.setState({ count: newCount });
-
-    this.props.updateQuery({
-      schedule: Object.assign({}, this.props.query.schedule, { interval: totalSeconds }),
-    });
+    this.newSchedule = { interval: totalSeconds };
   };
 
   setScheduleUntil = (momentDate, date) => {
-    this.props.updateQuery({
-      schedule: Object.assign({}, this.props.query.schedule, { until: date }),
-    });
+    this.newSchedule = { until: date };
   };
 
   setWeekday = (e) => {
     const dayOfWeek = e.target.value;
     this.setState({ dayOfWeek });
-    this.props.updateQuery({
-      schedule: Object.assign({}, this.props.query.schedule, {
-        day_of_week: dayOfWeek ? WEEKDAYS_FULL[WEEKDAYS_SHORT.indexOf(dayOfWeek)] : null,
-      }),
-    });
+    this.newSchedule = {
+      day_of_week: dayOfWeek ? WEEKDAYS_FULL[WEEKDAYS_SHORT.indexOf(dayOfWeek)] : null,
+    };
   };
 
   render() {
-    const schedule = this.props.query.schedule;
-    const format = 'YYYY-MM-DD';
-    const additionalAttributes = {};
+    const {
+      interval, minute, hour, until, count,
+    } = this.state;
 
     return (
       <Modal
         title="Refresh Schedule"
         className="schedule"
         visible={this.props.show}
         onCancel={this.props.onClose}
-        footer={[null, null]}
+        footer={null}
       >
         <div className="schedule-component">
           <div>Refresh every</div>
-          {schedule.interval ? (
-            <select value={this.state.count} onChange={this.setCount}>
-              {this.getAcceptableCounts().map(count => (
-                <option value={String(count)} key={count}>
-                  {String(count)}
-                </option>
+          {interval !== IntervalEnum.NEVER ? (
+            <select value={count} onChange={this.setCount}>
+              {this.counts.map(cnt => (
+                <option value={String(cnt)} key={cnt}>{cnt}</option>
               ))}
             </select>
           ) : null}
-          <select value={this.state.interval} onChange={this.setInterval}>
-            {getAcceptableIntervals(this.props.refreshOptions).map(iv => (
-              <option value={iv.name || ''} key={iv.name}>
-                {String(iv.name)}
-              </option>
+          <select value={interval} onChange={this.setInterval}>
+            {this.intervals.map(iv => (
```

Comment:
We need to take into account the case where there won't be any option for some interval. The allowed intervals is a setting, the admin can change and some might remove all the per minute intervals all together.

Reply:
Cool. I'll test it and open an issue if it falls short.

Reply:
@arikfr so you mean that memoization needs to depend on the interval settings?

Reply:
memoization? maybe I commented in the wrong place in the code 😳 but what I meant is that if we don't have any minute level intervals available, then we should offer "Minute" as one of the option in the dropdown.

Reply:
Nm. I'll make sure it works as expected 🤝

Reply:
Yeah it works as requested.

The only thing to take into consideration is that I put **memoization** on it (or else it reconstructs the array on any scheduler state change). This means that on the off-chance that these specific settings have changed for the user while a query is open, it would update only on re-entering/refreshing a query page.

Can either remove memoization, or leave as is @arikfr.

Reply:
Can leave as is -- this is not something that changes that frequently.